toolchains/kubectl/kubectl_configure.bzl

```diff
@@ -16,16 +16,79 @@ Defines a repository rule for configuring the kubectl tool.
+    else:
+        kubectl_tool_path = repository_ctx.which("kubectl")
+        substitutions = {"%{KUBECTL_TOOL}": "%s" % kubectl_tool_path}
+        template = Label("@io_bazel_rules_k8s//toolchains/kubectl:BUILD.path.tpl")
 
     repository_ctx.template(
         "BUILD",
-        Label("@io_bazel_rules_k8s//toolchains/kubectl:BUILD.tpl"),
-        {"%{KUBECTL_TOOL}": "%s" % kubectl_tool_path},
-        False
+        template,
+        substitutions,
+        False,
     )
 
-kubectl_configure = repository_rule(
+_kubectl_configure = repository_rule(
     implementation = _impl,
+    attrs = {
+        "build_srcs": attr.bool(
+            doc = "Optional. Set to true to build kubectl from sources.",
+            default = False,
+            mandatory = False,
+        ),
+    },
 )
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def kubectl_configure(name, **kwargs):
+    """
+    Creates an external repository with a kubectl_toolchain target
+    properly configured.
+
+    Args:
+        **kwargs:
+      Required Args
+        name: A unique name for this rule.
+      Default Args:
+        build_srcs: Optional. Set to true to build kubectl from sources. Default: False
+        k8s_commit: Otional. Commit / release tag at which to build kubectl
+          from. Default "v1.13.0-beta.1"
+        k8s_sha256: Otional. sha256 of commit at which to build kubectl from.
+          Default <valid sha for default version>.
+        k8s_prefix: Otional. Prefix to strip from commit / release archive.
+          Typically the same as the commit, or Kubernetes-<release tag>.
+          Default <valid prefix for default version>.
+      Note: Not all versions/commits of kubernetes project can be used to compile
+      kubectl from an external repo. Notably, we have only tested with v1.13.0-beta.1
+      or above. Note this rule has a hardcoded pointer to io_kubernetes_build repo
+      if your commit (above v1.13.0-beta.1) does not work due to problems,
+      related to @io_kubernetes_build repo, please send a PR to update these values.
+    """
+    build_srcs = False
+    if "build_srcs" in kwargs and kwargs["build_srcs"]:
+        build_srcs = True
+
+        # We keep these defaults here as they are only used by in this macro and not by the repo rule.
+        k8s_commit = kwargs["k8s_commit"] if "k8s_commit" in kwargs else "v1.13.0-beta.1"
+        k8s_sha256 = kwargs["k8s_sha256"] if "k8s_sha256" in kwargs else "dfb39ce36284c1ce228954ca12bf016c09be61e40a875e8af4fff84e116bd3a7"
+        k8s_prefix = kwargs["k8s_prefix"] if "k8s_prefix" in kwargs else "kubernetes-1.13.0-beta.1"
+        http_archive(
+            name = "io_kubernetes",
+            sha256 = k8s_sha256,
+            strip_prefix = k8s_prefix,
+            urls = [("https://github.com/kubernetes/kubernetes/archive/%s.tar.gz" % k8s_commit)],
+        )
+        http_archive(
```

Comment:
Do we need this?

Reply:
yes, this is what Jeff Grafton was referring to as problematic with the kubernetes repo and which would be solved by the abandoned cl.